plugin/data_masking/src/udf/udf_gen_dictionary_drop.cc

```diff
@@ -101,8 +101,10 @@ char *gen_dictionary_drop(UDF_INIT *, UDF_ARGS *args, char *result,
   DBUG_ENTER("gen_dictionary_drop");
 
   std::string res = _gen_dictionary_drop(args->args[0]);
-  *length = res.size();
-  strcpy(result, res.c_str());
+
+  strncpy(result, res.c_str(), *length - 1);
+  result[*length - 1] = '\0';
+  *length = std::min(res.size(), *length - 1);
 
```

Comment:
As per our slack discussions:
1. Change the logic to use initd->ptr because res has limited buffer allocated by caller/upper layers. We many need to return string bigger than the size of res buffer.
2. Expand gen_blacklist testcase to return huge string